tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -501,9 +501,44 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
                       CublasLtMatmulMaybeF8(&existing_gemm).WithOneUser())
                       .WithOneUser(),
                   m::Broadcast(&bias, m::Op())))) {
-      TF_ASSIGN_OR_RETURN(
-          bool was_fused,
-          FuseVectorBiasAdd(instr, bias, existing_gemm, optional_slice));
+      HloInstruction *optional_bias_f16 = nullptr;
+      if (existing_gemm->custom_call_target() == kCublasLtMatmulF8CallTarget &&
+          bias->shape().element_type() == F32) {
+        auto compatible_bias_type = [&](const HloInstruction *instr) -> bool {
+          auto gemm_d_type = existing_gemm->shape().element_type();
+          if (instr->shape().element_type() == BF16) {
+            return gemm_d_type == F8E4M3FN || gemm_d_type == F8E5M2 ||
+                   gemm_d_type == F32 || gemm_d_type == BF16;
+          } else if (instr->shape().element_type() == F16) {
+            return gemm_d_type == F16 || gemm_d_type == F8E4M3FN ||
+                   gemm_d_type == F8E5M2;
+          }
+          return false;
+        };
+
+        if (!Match(bias->mutable_operand(0),
+                   m::Convert(m::Convert(&optional_bias_f16, m::Op())
```

Comment:
Why are there two Converts here? Isn't just one necessary?

Reply:
`bias` here is actually `broadcast` and `bias->mutable_operand(0)` is the real bias of type f32. Then it's only possible to have a f16/bf16 bias in its operand.

Reply:
I understand `bias` is actually a broadcast, but why match two converts? You are matching `m::Convert(m::Convert(...)...)`, but it seems you only need to match `m::Convert(...)`. A user could have a single bf16->fp32 cast.